plugins/woocommerce-blocks/assets/js/blocks/product-collection/edit/inspector-controls/price-range-control/PriceTextField.tsx

```diff
@@ -0,0 +1,127 @@
+/* eslint-disable @wordpress/no-unsafe-wp-apis */
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { getCurrency } from '@woocommerce/price-format';
+import { Currency } from '@woocommerce/types';
+import {
+	// @ts-expect-error Using experimental features
+	__experimentalInputControl as InputControl,
+	// @ts-expect-error Using experimental features
+	__experimentalInputControlPrefixWrapper as InputControlPrefixWrapper,
+} from '@wordpress/components';
+
+interface PriceTextFieldProps {
+	value: number;
+	onChange: ( value: number | undefined ) => void;
+	label?: string;
+	suffix?: string;
+}
+
+const formatNumber = ( val: number, currency: Currency ): string => {
+	// Round the number to the correct number of decimals
+	const factor = Math.pow( 10, currency.minorUnit );
+	const roundedValue = `${ Math.round( val * factor ) / factor }`;
+
+	// Split the number into whole and decimal parts
+	let [ whole, decimal ] = roundedValue.split( '.' );
+
+	// Apply the thousand separator
+	whole = whole.replace(
+		/\B(?=(\d{3})+(?!\d))/g,
+		currency.thousandSeparator
+	);
+
+	// If there is no decimal part, we don't need to add decimal separator
+	if ( ! decimal ) {
+		return whole;
+	}
+
+	// Reassemble the number with the correct decimal separator
+	return currency?.decimalSeparator
+		? `${ whole }${ currency.decimalSeparator }${ decimal }`
+		: `${ whole }.${ decimal }`;
+};
+
+const formatValueWithCurrencySymbol = (
+	val: number | undefined,
+	currency: Currency
+) => {
+	if ( val === undefined || isNaN( val ) ) {
```

Comment:
Nitpicking, but `isNaN` covers also val being `undefined` so it can be simplified. However I understand if you want to keep it for explicitly showing the cases it's supposed to cover

```suggestion
	if ( isNaN( val ) ) {
```

Reply:
Actually, I included the `val === undefined` condition because removing it leads to a TypeScript error in the following line:

```ts
let formattedNumber = formatNumber(val, currency);
```

By explicitly adding this condition, I'm informing TypeScript that `val` will not be undefined 😀